internal/resources/aws/elasticache_cluster.go

```diff
@@ -104,3 +121,56 @@ func (r *ElastiCacheCluster) backupStorageCostComponent() *schema.CostComponent
 		},
 	}
 }
+
+// ElastiCache implementation of reservationResolver
+type elasticacheReservationResolver struct {
+	term          string
+	paymentOption string
+	cacheNodeType string
+}
+
+func newElasticacheReservationResolver(term, paymentOption, cacheNodeType string) reservationResolver {
+	return &elasticacheReservationResolver{
+		term:          term,
+		paymentOption: paymentOption,
+		cacheNodeType: cacheNodeType,
+	}
+}
```

Comment:
I don't see the value of these constructor functions. They just add noise. If you're not doing any additional functionality apart from just mapping terms to the fields, then let's remove it. If you're doing this constructor to return the interface, I would suggest against it. The general GO foo is ["accept interfaces return structs"](https://medium.com/@cep21/what-accept-interfaces-return-structs-means-in-go-2fe879e25ee8).

Reply:
Nice article! :) I removed the interface and so the not necessary extra complexity!